### PR TITLE
Work on the developers guide

### PR DESCRIPTION
**Pull request by RudolfWeeber**
This improves the sphinx version of the developrs guide.
New stuff on contacting developers, starting new development projects, cmake build system, doc structure, testsuite structure
Up-to-date version of adding bonded ia
Up-to-date version of global variables
Outdated infromation has been marked as such or removed entirely, depending on re-usability.

This pr also removes the LaTex version, which should no longer be used or changed.

There is much more to do at future coding days.

### CODE REVIEW

File: doc/sphinx/dg.rst [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
    The information found in this version of the Developer's Guide is
-   outdated.  Please see the section :ref:`getting_into_contact` and
+   outdated.  Please see the section :ref:contact` and
    ask for advice if you plan to start working on |es|.
 
-.. _getting_into_contact:
 
-Getting into Contact
-====================
+.. _contact:
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: Please use labels that match exactly the heading (except for the leading underscore).

File: doc/sphinx/dg.rst [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
-The first thing that you should do when you want to start to participate
-is to get into contact with the |es| developers. To do that, subscribe
-to the developers’ mailing list at
-http://lists.nongnu.org/mailman/listinfo/espressomd-devel and write a
-short email to the list address ``espressomd-devel@nongnu.org`` where
-you state your experience with simulations and programming and the
-things that you would like to do in |es|. We do not bite, and we are
-happy about anybody who wants to participate!
 
 .. _development_environment:
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: Same here, please correct the label.

File: doc/sphinx/dg.rst [hunk 3]
```diff
@@ -30,466 +34,99 @@ Development Environment
...
    git.
 
--  To build |es| from the development sources, you will need not too
-   old versions of the “GNU autotools” (*i.e.* automake [2]_ and
-   autoconf [3]_) installed on your system. Section
-   :ref:`building_the_development_code` contains information on how to
-   build the development code, section
-   :ref:`build_system` contains details on how the
-   build system works.
+-  The documentation is currently being converted form Latex to Sphinx. To build the old user and developer guides, you will need LaTex. For building tthe sphinx documentation, you will need the Python packages listed in ``requirements.txt`` in the top-level source directory. To install them, issue::
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: typo "from"

File: doc/sphinx/dg.rst [hunk 4]
```diff
@@ -30,466 +34,99 @@ Development Environment
...
-
-   -  ``clean``, ``distclean``
-
-.. _git_repositories:
-
-Git Repositories
-----------------
-
--  Development Repository: https://github.com/espressomd/espresso
+.. _getting_the_development_code:
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: Same here, please correct label.

File: doc/sphinx/dg.rst [hunk 5]
```diff
@@ -30,466 +34,99 @@ Development Environment
...
-
--  Check the distribution by using ``make distcheck``
+Adding New Source Files
+-----------------------
 
--  Add the files to the Git repository
+To add new files to |es| (like C++ source files or header files) you
+need to look at the CMakeList.txt in the directory where the file is located.
+* Please note that .hpp-header files usually do not have to be added to CMakeList.txt
+* In some cases (e.g., src/core/CMakeList.txt), the CMakeList.txt contains a wild-card include like this
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst:95]: Maybe we should mention that wildcards are in general a bad idea in CMakeLists.txt files.

File: doc/sphinx/dg.rst [hunk 6]
```diff
@@ -30,466 +34,99 @@ Development Environment
...
-   be quoted like ``\_``. On the other hand, it is possible to use other
-   layout commands (like ``\textit``) inside.
-
--  For pieces of Tcl-code that make extensive use of ``{`` and ``}``, a
-   verbatim environment ``tclcode`` exists:
-
-   ::
+The documentation of |es| consists of four parts:
+-  The users' guide and developers' guide are located in ``doc/sphinx``, and make use of the Sphinx Python package
+-  In-code documentation for the Python interface is located in the various files in src/python/espressomd and also makes use of the Sphinx Python package
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: Mentioning the numpydoc package and the respective doc style.

File: doc/sphinx/dg.rst [hunk 7]
```diff
@@ -535,129 +172,202 @@ description of the most common commands we need:
...
-After the setup of the necessary data structures in
-``interaction_data.h``, write the source file, something like
-``new_interaction.h``. You may want to use ``fene.h`` as a template
-file. Typically, you will have to define the following procedures:
-
--  ::
-
-       int *_set_params(int bond_type, ...)
+To add a new bonded interaction, the following steps have to be taken
+* Simulatino core:
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: typo "Simulation"

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

File: doc/sphinx/dg.rst (hunk 5 above)
Comment (fweik) [reply doc/sphinx/dg.rst:95]: I am not convinced that this is true.

**Comment by RudolfWeeber** [comment]
On Tue, May 09, 2017 at 07:59:42AM -0700, Florian Weik wrote:
> +To add new files to |es| (like C++ source files or header files) you
> +need to look at the CMakeList.txt in the directory where the file is located.
> +* Please note that .hpp-header files usually do not have to be added to CMakeList.txt
> +* In some cases (e.g., src/core/CMakeList.txt), the CMakeList.txt contains a wild-card include like this
> 
> I am not convinced that this is true.
Which statement exactly?
Concerning the wild cards: I'm quoting the part of the CMakeList
Concerning headers: apart from special cases like the feature
handling, they aren't for src/core and src/script_interface

**Comment by fweik** [comment]
No, the statement by @KaiSzuttor that wildcards are a bad idea.

**Comment by jonaslandsgesell** [comment]
Kai is probably refering to the need to rerun cmake when adding a new source file if the CMakeList is not modified. https://cmake.org/pipermail/cmake/2007-February/012833.html

**Review by fweik — APPROVED** [review]
Excellent

File: doc/sphinx/dg.rst [hunk 8]
```diff
@@ -7,7 +7,7 @@ Developer's Guide
    ask for advice if you plan to start working on |es|.
 
 
-.. _contact:
+.. _Contact the Develoeprs:
```
Comment (KaiSzuttor) [inline doc/sphinx/dg.rst]: You totally followed my remarks but I didn't meant to copy typos 😃

File: doc/sphinx/dg.rst [hunk 9]
```diff
@@ -1,495 +1,137 @@
 =================
 Developer's Guide
 =================
-
 .. warning::
    The information found in this version of the Developer's Guide is
-   outdated.  Please see the section :ref:`getting_into_contact` and
+   outdated.  Please see the section :ref:contact` and
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: `` :ref:`contact` `` has been renamed to `` :ref:`Contact_the_Developers` ``

File: doc/sphinx/dg.rst [hunk 10]
```diff
@@ -1,495 +1,137 @@
...
+
+
+.. _Development Environment:
 
 Development Environment
 =======================
 
-.. _required_development_tools:
+
+,, _Required Development Tools:
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: `.. _Required Development Tools:` (two dots, not two commas)

File: doc/sphinx/dg.rst [hunk 11]
```diff
@@ -1,495 +1,137 @@
...
 
 Required Development Tools
 --------------------------
-
-If you want to participate in the development of |es|, you will
-require the following tools depending on what tasks you want to perform:
+-  First of all, please install the dependencies for compiling |es|. See :ref:`_Getting, comping and running`
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: There is no label `` :ref:`_Getting, comping and running` ``

File: doc/sphinx/dg.rst [hunk 12]
```diff
@@ -1,495 +1,137 @@
...
    git.
 
--  To build |es| from the development sources, you will need not too
-   old versions of the “GNU autotools” (*i.e.* automake [2]_ and
-   autoconf [3]_) installed on your system. Section
-   :ref:`building_the_development_code` contains information on how to
-   build the development code, section
-   :ref:`build_system` contains details on how the
-   build system works.
+-  The documentation is currently being converted from Latex to Sphinx. To build the old user and developer guides, you will need LaTex. For building tthe sphinx documentation, you will need the Python packages listed in ``requirements.txt`` in the top-level source directory. To install them, issue::
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: Latex, LaTex?  It should be LaTeX.  Also: <strike>tthe</strike> the

File: doc/sphinx/dg.rst [hunk 13]
```diff
@@ -1,495 +1,137 @@
...
-.. _getting_into_contact:
 
-Getting into Contact
-====================
+.. _Contact the Develoeprs:
+
+Contact the Develoeprs
+======================
+
+To contact the |es| developers, please write an email to the developerss mailing list:
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: <strike>developerss</strike> developers

File: doc/sphinx/dg.rst [hunk 14]
```diff
@@ -1,495 +1,137 @@
...
--  To build |es| from the development sources, you will need not too
-   old versions of the “GNU autotools” (*i.e.* automake [2]_ and
-   autoconf [3]_) installed on your system. Section
-   :ref:`building_the_development_code` contains information on how to
-   build the development code, section
-   :ref:`build_system` contains details on how the
-   build system works.
+-  The documentation is currently being converted from Latex to Sphinx. To build the old user and developer guides, you will need LaTex. For building tthe sphinx documentation, you will need the Python packages listed in ``requirements.txt`` in the top-level source directory. To install them, issue::
+      pip install --user -r requirements.txt
+   Note, that some distributions nonw use ``pip`` for Python3 and ``pip2`` for Python 2.
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: <strike>nonw</strike> now

File: doc/sphinx/dg.rst [hunk 15]
```diff
@@ -1,495 +1,137 @@
...
-   :ref:`build_system` contains details on how the
-   build system works.
+-  The documentation is currently being converted from Latex to Sphinx. To build the old user and developer guides, you will need LaTex. For building tthe sphinx documentation, you will need the Python packages listed in ``requirements.txt`` in the top-level source directory. To install them, issue::
+      pip install --user -r requirements.txt
+   Note, that some distributions nonw use ``pip`` for Python3 and ``pip2`` for Python 2. 
 
--  To be able to compile the User’s Guide or the Developer’s Guide, you
-   will need a LaTeX-installation (all recent ones will do). For
-   details, refer to chapter .
+-  To build the tutorials, you wil need LaTex.
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: <strike>wil</strike> will.  LaTex should be LaTeX.

File: doc/sphinx/dg.rst [hunk 16]
```diff
@@ -1,495 +1,137 @@
...
--  install it into a new directory (``make install``)
+-  ``CMakeList.txt``
 
--  uninstall it (``make uninstall``)
+-  Contents of the ``cmake`` directory
 
-Whenever something goes wrong in these checks, it will give an error
-message that describes the problem. When everything goes fine, you can
-be relatively sure that you have a useful |es| distribution package.
+-  The CMakeList.txt files in the ``src/``, ``doc/``, and ``testsuite/`` directories and their sub-directories
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: <strike>CMakeList.txt</strike> CMakeLists.txt

File: doc/sphinx/dg.rst [hunk 17]
```diff
@@ -1,495 +1,137 @@
...
-
--  Run ``bootstrap.sh`` in the source directory
-
--  Check the distribution by using ``make distcheck``
+Adding New Source Files
+-----------------------
 
--  Add the files to the Git repository
+To add new files to |es| (like C++ source files or header files) you
+need to look at the CMakeList.txt in the directory where the file is located.
```
Comment (hmenke) [inline doc/sphinx/dg.rst:93]: <strike>CMakeList.txt</strike> CMakeLists.txt

File: doc/sphinx/dg.rst [hunk 18]
```diff
@@ -1,495 +1,137 @@
...
+* Please note that .hpp-header files usually do not have to be added to CMakeList.txt
```
Comment (hmenke) [inline doc/sphinx/dg.rst:94]: Remove "usually".  Header files *never* have to be added to CMakeLists.txt.  Only translation units have to be added.

File: doc/sphinx/dg.rst [hunk 19]
```diff
@@ -904,6 +604,34 @@ instead of ``return TCL_OK/TCL_ERROR`` you should use
...
+
+
+
+
+
+Global Variables which are synchronized across nodes
+----------------------------------------------------
+
+Adding new global variables to |es|, is strongly discuraged, because it means that code depends on a purely defined global state and cannot be tested individually.
+Features/Algorithms should instead be encapsulated in a class which is used by the script interface mechnaism.
```
Comment (hmenke) [inline doc/sphinx/dg.rst]: <strike>mechnaism</strike> mechanism

**Review by hmenke — CHANGES_REQUESTED** [review]

File: doc/sphinx/dg.rst [hunk 20]
```diff
@@ -533,131 +175,204 @@ description of the most common commands we need:
...
-.. _adding_new_nonbonded_interactions:
-
-Adding New Nonbonded Interactions
+To add a new bonded interaction, the following steps have to be taken
+* Simulation core:
+
+  * Define a structure holding the parameters (prefactors, etc.) of the interaction
+  * Write functions for calculating force and energy, respectively.
+  * Write a setter function, which takes the parameters of the interactions and stores them in the bonded interactions data structure
+  * Add calls to the force and energy calculatino functions to the force calculation in the integration loop as well as to energy and pressure/stress tensor analysis
```
Comment (pkreissl) [inline doc/sphinx/dg.rst]: [...] energy calculati**on** functions [...]

File: doc/sphinx/dg.rst [hunk 21]
```diff
@@ -533,131 +175,204 @@ description of the most common commands we need:
...
+      tensorial pressure calculation. For pair forces, the pressure is
+      calculated using the virials, for many body interactions currently no
+      pressure is calculated.
+   #  Do not forget to include the header file of your interaction.
+
+* Force calculation: in ``forces_inline.hpp` in the functino ``add_bonded_force()``, add your bond to the switch statement. For the FENE bond, e.g., the code looks like this::
+    case BONDED_IA_FENE:
+      bond_broken = calc_fene_pair_force(p1, p2, iaparams, dx, force);
+* Energy calculation: add similar code to ``add_bonded_energy()`` in ``energy_inline.hpp``
+* Pressur, stress tensor and virial calculation: If your bonded interaction is a pair bond and does not modify the particles involved, add simuilar code as above to presure.hpp:calc_bonded_pair_force(). Otherwise, you have to implement a custom solution for virial calculation.
```
Comment (pkreissl) [inline doc/sphinx/dg.rst]: Pressur**e** [...] sim<s>u</s>ilar [...]

File: doc/sphinx/dg.rst [hunk 22]
```diff
@@ -533,131 +175,204 @@ description of the most common commands we need:
...
+
+Adding the bonded interaciton in the Python interface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Pleas note that the following is Cython code (www.cython.org), rather than pure Python.
```
Comment (pkreissl) [inline doc/sphinx/dg.rst]: Pleas**e**

**Review by hmenke — APPROVED** [review]

**Pull request merged**